lightpath/ui/gui.py

```diff
@@ -173,6 +173,40 @@ def hidden_devices(self):
         return [dtype for button, dtype in self.device_buttons.items()
                 if not button.isChecked()]
 
+    def update_device_types(self):
+        """
+        Update the device type filter check boxes bases on devices
+        currently in the path
+        """
+        with self._lock:
+            # find valid device types. any() short circuits, yay generators!
+            valid_types = {}
+            for type_name, dtype in self.shown_types.items():
+                if any(isinstance(dev, dtype) for dev in self.path.path):
+                    valid_types[type_name] = dtype
+
+            # clear old checkboxes
+            for child in self._device_checkboxes:
+                self.device_types.layout().removeWidget(child)
+                self.device_buttons.pop(child)
+                child.deleteLater()
```

Comment:
I know there was talk of a double free-related segfault. I admittedly didn't pay enough attention to know if it could be related, but I see `deleteLater` here...

You could avoid deleting these by just creating all of the checkboxes and hiding the ones that aren't useful to avoid a create/remove/repeat

Reply:
I could make it so we never actually delete these checkboxes, rather we just store them and pick the ones we need.  

Is it generally bad to `deleteLater`?  I figured it'd help free up memory and guard against the strange off chance that we accumulate too many device types, but I could try to avoid it in general.

Reply:
If I hadn't vaguely recalled about the double free, I wouldn't have even commented. Did you by chance get to the bottom of that issue?

Normally `deleteLater` is just fine and isn't a reason on its own to suggest not using it.  There are scenarios where the overhead of creating/deleting/creating can be worse than just creating everything and hiding what you don't need, though it on a case-by-case basis where either method would be faster/more efficient/preferable.

Reply:
I poked around a bunch trying to reproduce the error, but to no avail.  

Just wanted to make sure I wasn't missing anything canonical w.r.t. pyqt gui development, and I'll certainly keep an eye out for the double free errors